source/com.microsoft.tfs.client.common/src/com/microsoft/tfs/client/common/credentials/CredentialsHelper.java

```diff
@@ -79,24 +79,29 @@ public static void refreshCredentialsForGit(final TFSConnection connection) {
             gitCredentialsManager.setCredentials(currentCredentials);
         }
 
-        if (cachedCredentials.equals(currentCredentials)) {
-            // The credentials haven't changed.
-            // No need to refresh.
-            return;
-        }
+        if (cachedCredentials != null) {
```

Comment:
No need to check for null here. We should return earlier if cachedCredentials == null